assets/src/components/editing/elements/common/settings/CaptionEditor.tsx

```diff
@@ -1,44 +1,29 @@
 import React from 'react';
 import { Caption, CaptionV2, ModelElement } from 'data/content/model/elements/types';
-import { getEditMode } from 'components/editing/elements/utils';
-import { useSlate } from 'slate-react';
 import { Model } from 'data/content/model/elements/factories';
 import { Descendant } from 'slate';
 import { CommandContext } from 'components/editing/elements/commands/interfaces';
-import { Editor } from 'components/editing/editor/Editor';
-import { blockInsertOptions } from 'components/editing/toolbar/editorToolbar/blocks/blockInsertOptions';
+import { InlineEditor } from './InlineEditor';
 
 const defaultCaption = (text = '') => [Model.p(text)];
-
 interface Props {
   onEdit: (caption: CaptionV2) => void;
   model: ModelElement & { caption?: Caption };
   commandContext: CommandContext;
 }
-export const CaptionEditor = (props: Props) => {
-  const editor = useSlate();
-  const editMode = getEditMode(editor);
 
+export const CaptionEditor: React.FC<Props> = ({ commandContext, model, onEdit }) => {
   return (
-    <div contentEditable={false}>
```

Comment:
Needed an editor for sub-parts of the definition.
Caption here was already doing this.
Refactored out the editor part of that to be usable in both situations.